lib/util.js

```diff
@@ -193,3 +195,52 @@ function encode_uri(uri) {
 }
 
 exports.encode_uri = encode_uri;
+
+function fetch_rss_entries (feed_url, max_items=100, max_days_old=7) {
+    const oldest_date = moment().subtract(max_days_old, 'days');
+
+    return _fetch_rss_page(feed_url, 1, [], max_items, oldest_date);
+}
+
+function _fetch_rss_json (feed_url) {
+    return new Promise((resolve, reject) => {
+        const req = request.get(feed_url);
+        const parser = new feedparser();
+        req.pipe(parser);
```

Comment:
hmm I would think that during execution of this block here the ops are still ordered so the `pipe` connecting would precede the handler for error which might not fire if something occurs between L209 and 211/212 executing. Maybe I'm missing something?